### PR TITLE
Pass callback’s return type through to transaction result

### PR DESCRIPTION
**Pull request by lumaxis**
I've been playing with this in a private fork recently and came to really like it so I wanted to share it upstream.
What I'm not completely sure about is how to merge/release this in a way that properly reflects the breaking nature of this change. Increasing the major version of the package is not really an option, I guess? Appreciate any help.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -5779,8 +5779,8 @@ declare module sequelize {
      * @param options Transaction Options
      * @param autoCallback Callback for the transaction
      */
-    transaction(options: TransactionOptions, autoCallback: (t: Transaction) => PromiseLike<any>): SequelizePromise<any>;
-    transaction(autoCallback: (t: Transaction) => PromiseLike<any>): SequelizePromise<any>;
+    transaction<TResult>(options: TransactionOptions, autoCallback: (t: Transaction) => PromiseLike<TResult>): SequelizePromise<TResult>;
+    transaction<TResult>(autoCallback: (t: Transaction) => PromiseLike<TResult>): SequelizePromise<TResult>;
```
Comment (felixfbecker) [inline index.d.ts]: Interesting, is this Sequelize runtime behaviour?

**Comment by felixfbecker** [comment]
Updates to typings can always be breaking, so that is nothing to worry about. But I don't think this is backwards-incompatible

File: index.d.ts (hunk 1 above)
Comment (lumaxis) [reply index.d.ts]: @felixfbecker Yup! That's actually exactly how we've been been using it already, just with the current `<any>` type.

**Comment by lumaxis** [comment]
@felixfbecker Happy to add a test! Do you have ideas how to do that here? Right now, `transaction` returns an `any` time and I'm not sure how I would test for a regression to that behavior 🤔

**Comment by lumaxis** [comment]
@felixfbecker Any ideas for how I could test this?

**Comment by felixfbecker** [comment]
you may be able to trigger a `noImplicitAny` error:
```ts
const result = sequelize.transaction(async () => [1, 2, 3])
result.forEach(x => console.log(x))
// or
const result = sequelize.transaction(async () => ({ foo: 123 }))
const foo = result.foo
```

**Comment by lumaxis** [comment]
Hm, didn't find an easy way to trigger the TypeScript error.
If you don't mind, I'd like to merge this as is for now.

**Pull request merged**